test/SlowTests/Issues/RavenDB-21693.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Threading.Tasks;
+using FastTests;
+using Raven.Server.Documents;
+using Sparrow.Server;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using static SlowTests.Issues.RavenDB_19033;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_21693 : NoDisposalNeeded
+    {
+        public RavenDB_21693(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [RavenFact(RavenTestCategory.Core)]
+        public async Task ShouldNotLockDatabaseForever()
+        {
+            var dbsCache = new ResourceCache<MyDb>();
+
+            var dbName = "foo";
+
+            var task1 = new Task<MyDb>(() =>
+            {
+                var myDb = new MyDb(dbName);
+
+                myDb.Run();
+
+                return myDb;
+            }, TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var database1 = dbsCache.GetOrAdd(dbName, task1);
+
+            if (database1 == task1)
+            {
+                task1.Start();
+                await task1;
+            }
+
+            var mre = new AsyncManualResetEvent();
+            var mre2 = new AsyncManualResetEvent();
+            Task t = null;
+            dbsCache.ForTestingPurposesOnly().OnUnlikelyTryGet = () =>
+            {
+                mre2.Set();
+                mre.WaitAsync().GetAwaiter().GetResult();
```

Comment:
I think we should create the `Task` with `TaskCreationOptions.LongRunning` option, so we will run this on a dedicated thread.
Otherwise we use a TP thread and might block other tasks

Reply:
oh, this is running on the xUnit thread? so better to make this test `void`

Reply:
Why, if it is `async void`, we cannot wait on it.

Reply:
not `async void` just `void`.. but actually I'm not sure how xUnit run those tests.

Reply:
done